packages/expo-maps/ios/AppleMapsView.swift

```diff
@@ -122,19 +173,7 @@ struct AppleMapsView: View, AppleMapsViewProtocol {
       .onChange(of: props.cameraPosition) { _, newValue in
         state.mapCameraPosition = convertToMapCamera(position: newValue)
       }
-      .onChange(of: state.selection) { _, newValue in
-        if let marker = props.markers.first(where: { $0.mapItem == newValue?.value }) {
-          props.onMarkerClick([
-            "title": marker.title,
-            "tintColor": marker.tintColor,
-            "systemImage": marker.systemImage,
-            "coordinates": [
-              "latitude": marker.coordinates.latitude,
-              "longitude": marker.coordinates.longitude
-            ]
-          ])
-        }
-      }
+      .onChange(of: state.selection, perform: handleSelectionChange)
```

Comment:
this was needed because xcode complained about the compiler not being able to type-check in reasonable time